applications/asset_tracker/src/main.c

```diff
@@ -348,6 +356,41 @@ static void cloud_cmd_handler(struct cloud_command *cmd)
 	/* Command handling goes here. */
 }
 
+#if CONFIG_BATTERY_MONITOR
+static void battery_monitor_check(struct k_work *work)
+{
+	static u8_t bat_charge;
+	static bool bat_level_acceptable = true;
+
+	battery_monitor_read(&bat_charge);
+
+	if (bat_charge < CONFIG_BATTERY_MONITOR_THRESHOLD &&
+	    bat_level_acceptable == true) {
+		ui_led_set_color(CONFIG_BATTERY_MONITOR_RED_VALUE,
+				 0,
+				 0,
+				 UI_LED_ON_PERIOD_CRITICAL,
+				 UI_LED_OFF_PERIOD_CRITICAL);
+		bat_level_acceptable = false;
+	} else if (bat_charge >= CONFIG_BATTERY_MONITOR_THRESHOLD &&
+	   bat_level_acceptable == false) {
+		ui_led_set_color(0,
```

Comment:
This breaks current pattern state set by ui_led_set_pattern() and is a bit confusing.
you could do ui_led_set_patttern(ui_led_get_pattern());